### DISCUSSION TITLE
Tracking with Timeseries

### DISCUSSION THREAD

**Discussion by Yuzal** [ravendb/ravendb#13022] [Q&A]
we have a application which is posting job ads.
we are currently tracking the views on each job ad by adding a new database entry in ravendb and we store several additional informations, for example remote ip, user agent, timestamps and the job add id.
quering this collection for reportings or statistics is quite heavy because we have more then 10m entries.
a while ago i saw a post about timeseries and was thinking if this could be a better solution for our case.
I am not entirely sure if this would be a better solution for tracking views with some additional informations because i need to store additional string fields and not numeric values.

Are the timeseries a better approach for our case?

**Comment by haludi** [comment]
Hi,

If the time series doesn't have numeric values at all I don't this you have
a reason you use it.
You can refer to a regular document by using the time series tag but that
is mostly relevant if most of the time you need the numeric values.
Also, it can be used if the additional information is repetitive.

Best regards,
Igal

**Comment by ayende** [comment]
In RavenDB 5.3 we have the incremental time series that are perfect for that scenario.

Se here:

https://ayende.com/blog/195267-C/ravendb-5-3-features-incremental-time-series?key=69dfe678d0a14788ae051cf0e0b77078

You'll be able to run this sort of code:

    session.IncrementalTimeSeriesFor("posts/2931-A", "INC:Views").Increment(1);


However, this is specifically for numeric values. The nice thing about it is that you can generate great reports very easily from this.

The question is whatever you cannot use multiple time series in this case?
You mentioned the Job Ad Id - isn't that just a time series _for that job ad_ ? In that case, you don't need to store that id. You are already storing that information in the time series attached to the job ad itself.

You can record multiple time series on the same transaction, and keep tracking in that manner.
